client/dfclient/include/resources.hpp

```diff
@@ -17,28 +17,35 @@ static const char* TEXTURES_PATH = "textures";
 static const char* LEVELS_PATH = "../levels";
 static const char* SOUNDS_PATH = "sounds";
 
+enum SoundType {
+	KILL,
+	DEATH,
+	GOLDFISH
+};
+
 struct Resources {
 	std::map<std::string, std::unique_ptr<ncine::Font>> fonts;
 	std::map<std::string, std::unique_ptr<ncine::Texture>> textures;
 
 	std::map<std::string, std::unique_ptr<ncine::AudioBuffer>> _sounds;
 	std::unique_ptr<ncine::AudioBuffer> _killSoundBuffer;
+	std::unique_ptr<ncine::AudioBuffer> _goldfishSoundBuffer;
 	std::unique_ptr<ncine::AudioBufferPlayer> _killSound;
 	std::unique_ptr<ncine::AudioBufferPlayer> _deathSound;
+	std::unique_ptr<ncine::AudioBufferPlayer> _goldfishSound;
 
-	std::vector<tweeny::tween<int>> _tweens;
+	std::vector<tweeny::tween<int>> _intTweens;
 	std::unordered_map<uint16_t, tweeny::tween<int>> _mobTweens;
+	std::vector<tweeny::tween<float>> _floatTweens;
+	std::optional<tweeny::tween<int>> _killingSpreeTween;
 
 	void UpdateTweens();
 
-	/** Plays default kill sound
+	/** Plays a sound
+	 * @param sound which sound type to play
 	 * @param gain volume (1.0f is 100%)
 	*/
-	void playKillSound(float gain = 1.0f);
-
-	/** Plays random death sound
-	*/
-	void playRandomDeathSound();
+	void playSound(SoundType soundType, float gain = 1.0f);
```

Comment:
Why are there sound changes on a killstreak pr? It's already huge man, have mercy

Reply:
xD Sry. My thought stream: I'm doing kill management -> goldfish kills are kills to be managed -> playing sounds is part of kill management -> doing a quick sound refactor was about as much work as sticking to the previous structure :D But you're right, it should go in a separate PR, like font_upgrade did (similar thing)